wagtail/admin/compare.py

```diff
@@ -297,6 +308,65 @@ def htmldiff(self):
         return mark_safe("\n".join(comparisons_html))
 
 
+class ListBlockComparison(StreamBlockComparison):
```

Comment:
I'm not keen on this being a subclass of StreamBlockComparison, and assuming that `super().get_block_comparisons()` is valid for use on ListBlock data - it might be now, but anyone updating `StreamBlockComparison.get_block_comparisons` in future has no way of knowing that they need to preserve compatibility with ListBlock and avoid any StreamBlock-only logic.

Could we give these a shared base class named something like `BaseSequenceBlockComparison`, maybe with the ID-based comparison logic in an appropriately named method like `get_block_comparisons_by_id` (which StreamBlockComparison.get_block_comparisons can then call unconditionally, and ListBlockComparison.get_block_comparisons can call after checking that usable IDs exist)?

Reply:
Thanks, that makes perfect sense. I've applied your suggestions.